src/HttpRequest.php

```diff
@@ -279,4 +282,19 @@ private function getServerVariable($key)
 
         return $this->server[$key];
     }
+
+    private function parseInputParameters()
+    {
+        if (isset($this->server['REQUEST_METHOD']) && isset($this->server['HTTP_CONTENT_TYPE']) &&
+            ($this->getMethod() == 'PUT' || $this->getMethod() == 'PATCH'))
+        {
+            if ($this->server['HTTP_CONTENT_TYPE'] == 'application/x-www-form-urlencoded') {
+                parse_str($this->inputStream, $this->postParameters);
+            }
+            else if ($this->server['HTTP_CONTENT_TYPE'] == 'application/json') {
+                $this->postParameters = json_decode($this->inputStream, true);
+            }
```

Comment:
json_decode does not belong in here. That is something unrelated to what this class is supposed to do.